java/gae-flexible-helloworld/src/main/java/com/example/bigtable/BigtableHelper.java

```diff
@@ -70,7 +74,19 @@ public static Connection getConnection() {
   public void contextInitialized(ServletContextEvent event) {
     // This will be invoked as part of a warmup request, or the first user
     // request if no warmup request was invoked.
-    sc = event.getServletContext();
+
+    if (PROJECT_ID != null && PROJECT_ID.startsWith("$")) PROJECT_ID = null;
```

Comment:
Technically this requires curly brackets. https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

I think you could achieve the same thing in a single line with a ternary operator.

```
PROJECT_ID = (PROJECT_ID != null && PROJECT_ID.startsWIth("$")) ? null : PROJECT_ID;
```